### PR TITLE
feat: Add modify_tables method

### PR DESCRIPTION
**Pull request by mure**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nisystemlink-clients-python/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

- Adds method for patching multiple tables in one call
- `BaseClient` takes in base path so clients don't have to manually build the URI
- Extracted simple table model to be reduce test verbosity

### What testing has been done?

- Added two automated tests for `modify_tables`

### CODE REVIEW

File: nisystemlink/clients/dataframe/models/_table_metadata_modification_base.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    state."""
+
+    name: Optional[str] = None
+    """The new name of the table. Setting to ``None`` will reset the name to the table's ID."""
+
+    workspace: Optional[str] = None
+    """The new workspace for the table. Setting to ``None`` will reset to the
+    default workspace. Changing the workspace requires permission to delete the
+    table in its current workspace and permission to create the table in its new
+    workspace."""
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_table_metadata_modification_base.py:24]: The "setting to None will reset" behavior is only applicable to the PATCH route. The multi-edit POST route considers null to mean unchanged as if it weren't present in the request.

File: nisystemlink/clients/dataframe/models/_modify_tables_partial_success.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+from typing import List
+
+from nisystemlink.clients.core import ApiError
+from nisystemlink.clients.core._uplink._json_model import JsonModel
+
+from ._modify_tables_request import TableMetdataModification
+
+
+class ModifyTablesPartialSuccess(JsonModel):
+    """The result of deleting multiple tables when one or more tables could not be deleted."""
```
Comment (spanglerco) [inline nisystemlink/clients/dataframe/models/_modify_tables_partial_success.py:10]:
```suggestion
    """The result of modifying multiple tables when one or more tables could not be modified."""
```

File: tests/integration/dataframe/test_dataframe.py [hunk 3]
```diff
@@ -20,7 +30,7 @@ def create_table(client: DataFrameClient):
     tables = []
 
     def _create_table(table: models.CreateTableRequest) -> str:
```
Comment (spanglerco) [inline tests/integration/dataframe/test_dataframe.py:32]: Given the `or basic_table_model` below, did you intend to make `table` optional?

```suggestion
    def _create_table(table: Optional[models.CreateTableRequest] = None) -> str:
```

(with an import being required)

**Review by spanglerco — CHANGES_REQUESTED** [review]

File: nisystemlink/clients/dataframe/models/_table_metadata_modification_base.py (hunk 1 above)
Comment (mure) [reply nisystemlink/clients/dataframe/models/_table_metadata_modification_base.py:24]: Yeah I realized that as I was writing the tests, but I didn't think about the docstrings. Maybe this base class isn't worth it.

Comment (mure) [reply nisystemlink/clients/dataframe/models/_table_metadata_modification_base.py:24]: Why is the behavior different anyway?

**Review by spanglerco — APPROVED** [review]

**Pull request merged**